devtools/common/src/main/java/io/quarkus/cli/commands/AddExtensions.java

```diff
@@ -33,35 +36,47 @@ public boolean addExtensions(final Set<String> extensions) throws IOException {
         List<Dependency> dependenciesFromBom = getDependenciesFromBom();
 
         for (String dependency : extensions) {
-            Optional<Extension> optional = MojoUtils.loadExtensions().stream()
+            List<Extension> matches = MojoUtils.loadExtensions().stream()
                     .filter(d -> {
                         boolean hasTag = d.labels().contains(dependency.trim().toLowerCase());
                         boolean machName = d.getName()
                                 .toLowerCase()
                                 .contains(dependency.trim().toLowerCase());
-                        return hasTag || machName;
+                        boolean matchArtifactId = d.getArtifactId()
+                                .toLowerCase()
+                                .contains(dependency.trim().toLowerCase());
+                        return hasTag || machName || matchArtifactId;
                     })
-                    .findAny();
+                    .collect(Collectors.toList());
+
+            if (matches.size() > 1) {
+                StringBuilder sb = new StringBuilder();
+                sb.append(NOK)
+                        .append(" Multiple extensions matching '" + dependency + "'");
 
-            if (optional.isPresent()) {
-                final Extension extension = optional.get();
+                matches.stream()
+                        .forEach(extension -> sb.append("\n     * ").append(extension.managementKey()));
```

Comment:
Does this need to use `java.lang.System#lineSeparator`?

Reply:
Thanks. Done.